x/authz/types/msgs.go

```
@@ -0,0 +1,188 @@
+func (msg MsgGrantAuthorization) UnpackInterfaces(unpacker types.AnyUnpacker) error {
+	var authorization Authorization
+	return unpacker.UnpackAny(msg.Authorization, &authorization)
+}
+
+// NewMsgRevokeAuthorization creates a new MsgRevokeAuthorization
+//nolint:interfacer
+func NewMsgRevokeAuthorization(granter sdk.AccAddress, grantee sdk.AccAddress, methodName string) MsgRevokeAuthorization {
+	return MsgRevokeAuthorization{
+		Granter:    granter.String(),
+		Grantee:    grantee.String(),
+		MethodName: methodName,
+	}
+}
+
+// GetSigners implements Msg
+func (msg MsgRevokeAuthorization) GetSigners() []sdk.AccAddress {
+	granter, err := sdk.AccAddressFromBech32(msg.Granter)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{granter}
+}
+
+// ValidateBasic implements MsgRequest.ValidateBasic
+func (msg MsgRevokeAuthorization) ValidateBasic() error {
+	if msg.Granter == "" {
+		return sdkerrors.Wrap(ErrInvalidGranter, "missing granter address")
+	}
+	if msg.Grantee == "" {
+		return sdkerrors.Wrap(ErrInvalidGrantee, "missing grantee address")
+	}
+	return nil
+}
+
+// NewMsgExecAuthorized creates a new MsgExecAuthorized
+//nolint:interfacer
+func NewMsgExecAuthorized(grantee sdk.AccAddress, msgs []sdk.ServiceMsg) MsgExecAuthorized {
+	msgsAny := make([]*types.Any, len(msgs))
+	for i, msg := range msgs {
+		bz, err := proto.Marshal(msg.Request)
+		if err != nil {
+			panic(err)
+		}
+
+		anyMsg := &types.Any{
+			TypeUrl: msg.MethodName,
+			Value:   bz,
+		}
+
+		msgsAny[i] = anyMsg
+	}
+
+	return MsgExecAuthorized{
+		Grantee: grantee.String(),
+		Msgs:    msgsAny,
+	}
+}
+
+// GetServiceMsgs returns the cache values from the MsgExecAuthorized.Msgs if present.
+func (msg MsgExecAuthorized) GetServiceMsgs() ([]sdk.ServiceMsg, error) {
+	msgs := make([]sdk.ServiceMsg, len(msg.Msgs))
+	for i, msgAny := range msg.Msgs {
+		msg1 := sdk.ServiceMsg{
+			MethodName: msgAny.TypeUrl,
+			Request:    msgAny.GetCachedValue().(sdk.MsgRequest),
```

Let's maybe return an error here, I see a potential panic.